### PR TITLE
Add Sorting filter on Discharged Patients page

### LINKED ISSUE #7581: Discharged patient's timeline order issue in facility's view patients feature
**Describe the bug**
Although patient's admission and discharged timestamp order (newest to oldest) is correct under both "active and discharged" tab in "patients" section in Home page; But in "view patients" section in a particular facility the order is coming as "oldest to newest" under "discharged" tab.

**To Reproduce**

1. Go to 'care.coronasafe.in'
2. Click on 'click on any facility's "view patients" button.'
3. Click on 'discharged' tab.
4. Order is wrong.

**Expected behavior**
1. order should be "newest to oldest" or
2. filter should be available to sort out data as per admission date timestamp.

### PR DESCRIPTION
**Pull request by Pranshu1902**
## Proposed Changes

- Fixes #7581 
Add a Sorting filter on the Discharged Patients page like that on Live patients page

Backend PR: [#2074](https://github.com/coronasafe/care/pull/2074)
image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/DischargedPatientsList.tsx [hunk 1]
```diff
@@ -24,7 +30,11 @@ const DischargedPatientsList = ({
     pathParams: { id: facility_external_id },
   });
 
-  const [search, setSearch] = useQueryParams();
+  const { qParams, updateQuery, FilterBadges } = useFilters({});
+
+  useEffect(() => {
+    updateQuery({ ordering: "-modified_date" });
+  }, []);
```
Comment (rithviknishad) [inline src/Components/Facility/DischargedPatientsList.tsx]:
```suggestion
```

File: src/Components/Facility/DischargedPatientsList.tsx [hunk 2]
```diff
@@ -51,10 +66,13 @@ const DischargedPatientsList = ({
...
       }
     >
+      <div className="col-span-3 mt-6 flex flex-wrap">
+        <FilterBadges badges={({ ordering }) => [ordering()]} />
+      </div>
       <PaginatedList
         route={routes.listFacilityDischargedPatients}
         pathParams={{ facility_external_id }}
-        query={search}
+        query={qParams}
```
Comment (rithviknishad) [inline src/Components/Facility/DischargedPatientsList.tsx]:
```suggestion
        query={{ordering: "-modified_date", ...qParams}}
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Comment by nihal467** [comment]
image

- [ ] keep sort by and live discharge switching button right aligned
- [ ] swap the places in between the button

**Comment by Pranshu1902** [comment]
@nihal467 Updated view:
image

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**